pkg/pod/container.go

```diff
@@ -235,6 +235,32 @@ func (builder *ContainerBuilder) WithEnvVar(name, value string) *ContainerBuilde
 	return builder
 }
 
+// WithVolumeMount adds a pod volume mount inside the container.
+func (builder *ContainerBuilder) WithVolumeMount(volMount v1.VolumeMount) *ContainerBuilder {
+	glog.V(100).Infof("Adding VolumeMount to the %s container's definition", builder.definition.Name)
+
+	if volMount.Name == "" {
+		glog.V(100).Infof("Container's VolumeMount name cannot be empty")
```

Comment:
BTW,  we're able technically to mount a volume inside a container without a name. An arbitrary name is generated in such cases. Do we still want to fail here?

Reply:
Thanks @achuzhoy  - unfortunately I've no idea where is that arbitrary name generate(by `oc` client or by some resource schema validation function on the server side, etc) so cannot give U a definitive answer.
I prefer to have it state clearly but if U disagree and  insist on allowing it to be empty - please let me know

Reply:
@achuzhoy There must be some linkage between pod.WithVolume, deployment.WithVolume and this method. I think that linkage is the name alone